internal/backends/wasm/wasm.go

```diff
@@ -0,0 +1,223 @@
+	FilenamePatterns: []string{"*.wasm"},
+	Quirks:           api.QuirksAddRemoveAlsoLocks,
+	GetPackageDir: func() string {
+		path := string(util.GetCmdOutput([]string{"wapm", "bin"}))
+		path = strings.TrimSuffix(path, "\n")
+		path = strings.TrimSuffix(path, "/.bin")
+		if path == "" {
+			return "wapm_packages"
+		}
+		return path
+	},
+	Search: func(query string) []api.PkgInfo {
+		client := graphql.NewClient("https://registry.wapm.io/graphql")
+		req := graphql.NewRequest(`
+        query searchPackages($query: String!) {
+            search(query: $query) {
+                edges {
+                    node {
+                        ...on PackageVersion {
+                            package {
+                                name
+                            }
+                            version
+                            description
+                            homepage
+                            repository
+                            license
+                            publishedBy {
+                                username
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        `)
+		// set any variables
+		req.Var("query", string(query))
+		ctx := context.Background()
+
+		var search searchQuery
+		if err := client.Run(ctx, req, &search); err != nil {
+			util.Die("WAPM response: %s", err)
+		}
+
+		results := []api.PkgInfo{}
+		for _, edge := range search.Search.Edges {
+			results = append(results, api.PkgInfo{
+				Name:        edge.Node.Package.Name,
+				Description: edge.Node.Description,
+				Version:     edge.Node.Version,
+				HomepageURL: edge.Node.Homepage,
+				Author:      edge.Node.PublishedBy.Username,
+				License:     edge.Node.License,
+			})
+		}
+		return results
+	},
+	Info: func(name api.PkgName) api.PkgInfo {
+		client := graphql.NewClient("https://registry.wapm.io/graphql")
+		req := graphql.NewRequest(`
+        query getPackage($name: String!) {
+            packageVersion: getPackageVersion(name: $name) {
+                package {
+                    name
+                }
+                version
+                description
+                homepage
+                repository
+                license
+                publishedBy {
+                    username
+                }
+            }
+        }
+        `)
+		// set any variables
+		req.Var("name", string(name))
+		ctx := context.Background()
+
+		var output infoQuery
+		if err := client.Run(ctx, req, &output); err != nil {
+			util.Die("WAPM response: %s", err)
+		}
+
+		return api.PkgInfo{
+			Name:        output.PackageVersion.Package.Name,
+			Description: output.PackageVersion.Description,
+			Version:     output.PackageVersion.Version,
+			HomepageURL: output.PackageVersion.Homepage,
+			Author:      output.PackageVersion.PublishedBy.Username,
+			License:     output.PackageVersion.License,
+		}
+	},
+	Add: func(pkgs map[api.PkgName]api.PkgSpec) {
```

Comment:
If I'm understanding this correctly:
`wapm init -y` creates a lock file (with nothing in it?)
`wapm add` will add things to the specfile (ignoring the lockfile?)
`wapm install` then installs the packages on disk and adds them to the specfile.

If this is the case you should be able to omit the `wapm install` at the end and remove the `QuirksAddRemoveAlsoInstalls` quirk.

Reply:
1. wapm init -y` creates a specfile file (no lockfile yet)
2. `wapm add` will add things to the specfile (ignoring the lockfile)
3. `wapm install` then installs the packages on specfile and adds them to the lockfile.

Reply:
yup, that sounds good.
Could you also stat the specfile instead of the lockfile when determining if `wapm init -y` should be run.

Reply:
> Could you also stat the specfile instead of the lockfile when determining if wapm init -y should be run.

`wapm init -y` is a no-op if `wapm.toml` already exists (similarly to npm). So we should be all good running it :)

Reply:
wait, so why even check if `wapm.lock` exists?